target/linux/ramips/dts/mt7621_asus_rt-n56u-b1.dts

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "asus,rt-n56u-b1", "mediatek,mt7621-soc";
+	model = "ASUS RT-N57U B1";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "blue:power";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan2g {
+			label = "blue:wlan2g";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0radio";
+		};
+
+		wlan5g {
+			label = "blue:wlan5g";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1radio";
+		};
+
+		wan {
+			label = "blue:wan";
+			gpios = <&gpio 7 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "dsa-0.0:04:link";
```

Comment:
Does this actually work?

Reply:
This is really work. I connect Ethernet cable to WAN port and led WAN on. If i disconnect cable, led is off

Reply:
I can't find solutions for LAN led.
Router have 4 LAN ports and only one LAN led. I set in current DT only if LAN1 port link up led on.
In stock firmware led on if any of 4ports link up

Reply:
I think user can set himself configuration for this led by netdev or something else

Reply:
cat /sys/class/leds/blue\:lan/trigger 
[none] timer default-on netdev dsa-0.0:00:link dsa-0.0:00:1Gbps dsa-0.0:00:100Mbps dsa-0.0:00:10Mbps dsa-0.0:01:link dsa-0.0:01:1Gbps dsa-0.0:01:100Mbps dsa-0.0:01:10Mbps dsa-0.0:02:link dsa-0.0:02:1Gbps dsa-0.0:02:100Mbps dsa-0.0:02:10Mbps dsa-0.0:03:link dsa-0.0:03:1Gbps dsa-0.0:03:100Mbps dsa-0.0:03:10Mbps dsa-0.0:04:link dsa-0.0:04:1Gbps dsa-0.0:04:100Mbps dsa-0.0:04:10Mbps heartbeat usbport phy0rx phy0tx phy0assoc phy0radio phy0tpt phy1rx phy1tx phy1assoc phy1radio phy1tpt

Reply:
cat /sys/class/leds/blue\:wan/trigger 
none timer default-on netdev dsa-0.0:00:link dsa-0.0:00:1Gbps dsa-0.0:00:100Mbps dsa-0.0:00:10Mbps dsa-0.0:01:link dsa-0.0:01:1Gbps dsa-0.0:01:100Mbps dsa-0.0:01:10Mbps dsa-0.0:02:link dsa-0.0:02:1Gbps dsa-0.0:02:100Mbps dsa-0.0:02:10Mbps dsa-0.0:03:link dsa-0.0:03:1Gbps dsa-0.0:03:100Mbps dsa-0.0:03:10Mbps [dsa-0.0:04:link] dsa-0.0:04:1Gbps dsa-0.0:04:100Mbps dsa-0.0:04:10Mbps heartbeat usbport phy0rx phy0tx phy0assoc phy0radio phy0tpt phy1rx phy1tx phy1assoc phy1radio phy1tpt

Reply:
> Router have 4 LAN ports and only one LAN led. I set in current DT only if LAN1 port link up led on.
> In stock firmware led on if any of 4ports link up

So, it's _not_ working?

Reply:
It's working bun not as expected. Ok i can make it without trigger (remove it).
User can make himself LED Configuration.
```
LED Name: blue:lan
Trigger: netdev
Device: br-lan
Trigger Mode:   Link On, Transmit, Receive
```
```
LED Name: blue:wan
Trigger: netdev
Device: wan
Trigger Mode:   Link On, Transmit, Receive
```
And all working good as factory firmware.



▾